modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/BinaryTupleComparator.java

```diff
@@ -15,80 +15,66 @@
 import org.apache.ignite.internal.storage.index.SortedIndexDescriptor.ColumnDescriptor;
 
 /**
- * Comparator implementation for comparting {@link BinaryTuple}s on a per-column basis.
+ * Comparator implementation for comparing {@link BinaryTuple}s on a per-column basis.
  */
-class BinaryTupleComparator implements Comparator<BinaryTuple> {
+public class BinaryTupleComparator implements Comparator<ByteBuffer> {
     private final SortedIndexDescriptor descriptor;
 
-    private final int prefixLength;
-
-    private BinaryTupleComparator(SortedIndexDescriptor descriptor, int prefixLength) {
-        if (prefixLength > descriptor.indexColumns().size()) {
-            throw new IllegalArgumentException("Invalid prefix length: " + prefixLength);
-        }
-
-        this.descriptor = descriptor;
-        this.prefixLength = prefixLength;
-    }
-
     /**
      * Creates a comparator for a Sorted Index identified by the given descriptor.
      */
-    static BinaryTupleComparator newComparator(SortedIndexDescriptor descriptor) {
-        return new BinaryTupleComparator(descriptor, descriptor.indexColumns().size());
-    }
-
-    /**
-     * Similar to {@link #newComparator} but creates a comparator that only compares first {@code prefixLength} index columns.
-     */
-    static BinaryTupleComparator newPrefixComparator(SortedIndexDescriptor descriptor, int prefixLength) {
-        return new BinaryTupleComparator(descriptor, prefixLength);
+    public BinaryTupleComparator(SortedIndexDescriptor descriptor) {
+        this.descriptor = descriptor;
     }
 
     @Override
-    public int compare(BinaryTuple tuple1, BinaryTuple tuple2) {
-        return compare(tuple1, tuple2, 1, 0);
-    }
+    public int compare(ByteBuffer buffer1, ByteBuffer buffer2) {
+        assert buffer1.order() == ByteOrder.LITTLE_ENDIAN;
+        assert buffer2.order() == ByteOrder.LITTLE_ENDIAN;
+        assert !(isPrefix(buffer1) && isPrefix(buffer2));
 
-    /**
-     * Compares a given tuple with the configured prefix.
-     *
-     * @param tuple1 Tuple to compare.
-     * @param tuple2 Tuple to compare.
-     * @param direction Sort direction: {@code -1} means sorting in reversed order, {@code 1} means  sorting in the natural order.
-     * @param equals Value that should be returned if the provided tuple exactly matches the prefix.
-     * @return the value {@code 0} if the given row starts with the configured prefix;
-     *         a value less than {@code 0} if the row's prefix is smaller than the prefix; and
-     *         a value greater than {@code 0} if the row's prefix is larger than the prefix.
-     */
-    public int compare(BinaryTuple tuple1, BinaryTuple tuple2, int direction, int equals) {
-        for (int i = 0; i < prefixLength; i++) {
+        BinaryTupleSchema schema = descriptor.binaryTupleSchema();
+
+        InternalTuple tuple1 = isPrefix(buffer1) ? new BinaryTuplePrefix(schema, buffer1) : new BinaryTuple(schema, buffer1);
```

Comment:
I think `isPrefix` can be extracted into a variable (you already called this method in the assert statement)